core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java

```diff
@@ -35,6 +37,16 @@ public abstract class TaskRunner implements Plugin, WorkerJobLifecycle {
     @JsonIgnore
     @Getter(AccessLevel.NONE)
     private transient Map<String, String> env;
+    
+    @JsonIgnore
+    @Builder.Default
+    @Getter(AccessLevel.NONE)
+    private List<Runnable> killable = new ArrayList<>();
```

Comment:
I'm not sure it's worth the complexity of registering multiple runnable. I cannot see any usage.
I think having a kill() method on TaskRunner would be enough and more "declarative"